packages/vector_graphics/lib/src/listener.dart

```diff
@@ -41,25 +44,31 @@ Future<PictureInfo> decodeVectorGraphics(
   ByteData data, {
   required Locale? locale,
   required TextDirection? textDirection,
+  required BytesLoader loader,
 }) async {
-  assert(() {
-    _debugLastTextDirection = textDirection;
-    _debugLastLocale = locale;
-    return true;
-  }());
-  final FlutterVectorGraphicsListener listener = FlutterVectorGraphicsListener(
-    locale: locale,
-    textDirection: textDirection,
-  );
-  DecodeResponse response = _codec.decode(data, listener);
-  if (response.complete) {
+  try {
+    assert(() {
+      _debugLastTextDirection = textDirection;
+      _debugLastLocale = locale;
+      return true;
+    }());
+    final FlutterVectorGraphicsListener listener =
+        FlutterVectorGraphicsListener(
+      locale: locale,
+      textDirection: textDirection,
+    );
+    DecodeResponse response = _codec.decode(data, listener);
+    if (response.complete) {
+      return listener.toPicture();
+    }
+    await listener.waitForImageDecode();
+    response = _codec.decode(data, listener, response: response);
+    assert(response.complete);
+
     return listener.toPicture();
+  } catch (e) {
```

Comment:
Can we catch exception instead of everything? As written this will catch things like NPEs, which should really just crash so we can fix it...

Reply:
The thrown exception contains the original exception.

I want this because when dealing with an internal roll that used an invalid vector graphic binary, I couldn't tell which one was failing - only that something was causing a cast error in the codec.

Reply:
Makes sense!

Reply:
(IOW, this is basically a catch and rethrow, just not using rethrow explicitly so I can add extra information)